apps/examples/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,459 @@
+			return ERROR;
+		}
+
+		close(fd);
+
+		ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+		ioctl(frt_fd, TCIOC_STOP, 0);
+
+		close(frt_fd);
+
+		tt  = after.timeleft - before.timeleft;
+		printf("Total time taken is: %lu us\n", tt);
+
+		ret = unlink(file[0]);
+		if (ret != OK) {
+			printf("Failed to unlink file %s, ret : %d, errno : %d\n", file[0], ret, errno);
+			return ERROR;
+		}
+		break;
+
+	case TEST_FLAG_OVERWRITE :
+		memset(rw_buf, 0, BUFFER_SIZE);
+		tt = tt2 = 0;
+		printf(" File size : %d bytes, Buffer size : %d bytes\n", f_sz, b_sz);
+
+		fd = open(file[0], O_WROK | O_CREAT);
+		if (fd < 0) {
+			printf("Unable to do open file : %s,  fd : %d\n", file[0], fd);
+			return ERROR;
+		}
+
+		for (i = 0; i < (f_sz / b_sz); i++) {
+			ret = write(fd, rw_buf, b_sz);
+			if (ret != (b_sz)) {
+				printf("Unable to write in file: %s, size : %d, ret : %d\n", file[0],b_sz, ret);
+				close(fd);
+				return ERROR;
+			}
+		}
+		close(fd);
+
+		for (i = 1; i <= 5 ; i++) {
+			frt_fd = open("/dev/timer0", O_RDONLY);
+			if (frt_fd < 0) {
+				printf("Unable to do open file : /dev/timer0, errno : %d\n", file[0], errno);
+				return ERROR;
+			}
+			ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN);
+			ioctl(frt_fd, TCIOC_START, TRUE);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before);
+
+			fd = open(file[0], O_WROK);
+			if (fd < 0) {
+				printf("Unable to do open file : %s,  fd : %d\n", file[0], fd);
+				close(frt_fd);
+				return ERROR;
+			}
+
+			ret = lseek(fd, 0, SEEK_SET);
+			if (ret != 0) {
+				printf("Unable to move file pointer to start\n");
+				close(frt_fd);
+				close(fd);
+				return ERROR;
+			}
+
+			for (int i_wr = 1; i_wr <= (f_sz / b_sz); i_wr++) {
+				ret = write(fd, rw_buf, b_sz);
+				if (ret != (b_sz)) {
+					printf("Unable to write in file: %s, iter= %d ,ret = %d\n", file[0], i_wr, ret);
+					close(frt_fd);
+					close(fd);
+					return ERROR;
+				}
+			}
+
+			close(fd);
+			ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after);
+			ioctl(frt_fd, TCIOC_STOP, 0);
+			close(frt_fd);
+
+			tt2 = after.timeleft - before.timeleft;
+			tt += tt2;
+			printf(" iteration : %d, time : %lu\n", i, tt2);
+		}
+		printf("total time :  %lu us\n", tt);
+
+		ret = unlink(file[0]);
+		if (ret != OK) {
+			printf("Failed to unlink file %s, ret : %d, errno : %d\n", file[0], ret, errno);
+			return ERROR;
+		}
+		break;
+
+	case TEST_FLAG_APPEND :
+		memset(rw_buf, 0, BUFFER_SIZE);
+		tt = tt2 = 0;
+		for(int it = 0; it <= 5; it++) {
+			fd = open(file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file : %s, fd = %d\n", file[0], fd);
+				return ERROR;
+			}
+			/*create a test file of size f_sz */
+			ret = write(fd, rw_buf, f_sz);
+			if (ret != (f_sz)) {
+				printf("Unable to write in file: %s, size = %d, ret = %d\n", file[0], f_sz, ret);
+				close(fd);
+				return ERROR;
+			}
+                	close(fd);
+
+			frt_fd = open("/dev/timer0", O_RDONLY);
```

Comment:
Timer device number depends on the device this app is running on
Timer1 for rtk
Timer0 for mtk